src/main/java/org/springframework/integration/kafka/inbound/KafkaMessageDrivenChannelAdapter.java

```diff
@@ -63,11 +77,49 @@ public void setAutoCommitOffset(boolean autoCommitOffset) {
 		this.autoCommitOffset = autoCommitOffset;
 	}
 
+	/**
+	 * Generate Message Ids for produced messages. If set to false, will try to use a default value. By default set to false.
+	 * Please note that this option only works in conjunction with {@link #setUseMessageBuilderFactory(boolean)}. If the
+	 * latter is set to true, then some {@link MessageBuilderFactory} implementations such as
+	 * {@link DefaultMessageBuilderFactory} may ignore it.
+	 *
+	 * @param generateMessageId true if a message id should be generated
+	 */
+	public void setGenerateMessageId(boolean generateMessageId) {
+		this.generateMessageId = generateMessageId;
+	}
+
+	/**
+	 * Generate timestamp for produced messages. If set to false, -1 is used instead. By default set to false.
+	 * Please note that this option only works in conjunction with {@link #setUseMessageBuilderFactory(boolean)}. If the
+	 * latter is set to true, then some {@link MessageBuilderFactory} implementations such as
+	 * {@link DefaultMessageBuilderFactory} may ignore it.
+	 *
+	 * @param generateTimestamp true if a timestamp should be generated
+	 */
+	public void setGenerateTimestamp(boolean generateTimestamp) {
+		this.generateTimestamp = generateTimestamp;
+	}
+
+	/**
+	 * Use the {@link MessageBuilderFactory} returned by {@link #getMessageBuilderFactory()} to create messages.
+	 *
+	 * @param useMessageBuilderFactory true if the @link MessageBuilderFactory} returned by
```

Comment:
OK. No problem: https://jira.spring.io/browse/INTEXT-154 !
I'm merging it with those polishing